waspc/ChangeLog.md

```diff
@@ -2,6 +2,41 @@
 
 ## v0.9.1
 
+### Adds support for database seeding
+You can now define JS/TS functions for seeding the database!
+
+```c
+app MyApp {
+  // ...
+  db: {
+    seeds: [
+      import { devSeedSimple } from "@server/dbSeeds.js",
+      import { prodSeed } from "@server/dbSeeds.js",
+    ]
+  }
+}
+```
+
+```js
+import { createTask } from './actions.js'
+
+export const devSeedSimple = async (prismaClient) => {
+  const { password, ...newUser } = await prismaClient.user.create({
+    username: "RiuTheDog", password: "bark1234"
+  })
+  await createTask(
+    { description: "Chase the cat" },
+    { user: newUser, entities: { Task: prismaClient.task } }
+  )
+}
+
+//...
+```
+
+`wasp db seed` runs the first(default) seed, while `wasp db seed <name>` will run a seed with specific name.  
+For example, for the case above, you could do `wasp db seed prodSeed`.
```

Comment:
You are right! I just wanted to make it simple, but you make a good point, it is certianly not intuitive and it makes you think all of them will run. I will play a bit with it in the direction of what you suggested and make it more intuitive!

Reply:
I have been looking more into it now, and there is one thing -> the list is simple.
In the sense that they don't have to think much about when specifying seeds.
Yes, only first will run by default, and that is not obvious, but it is really easy to specify seeds.

While if we have `default` and `others`, they have to start immediatelly thinking about where to put and how to organize the seeds. Also `others` is a bit weird name, not terrible, but also not great, and I can't come up with a better name.

I will try writing down a couple of approaches:
1.
```
seeds: [<extimport>]
```
This is what we have now. It is not clear that first one is default, but that is kind of  "a happy little accident". It does make defining them really simple, not much thinking.

2.
I could do she same thing as (1), just remove the idea of default seed -> you always have to specify the name of the seed in `wasp db seed <name>`. DSL is kept simple, the idea of "default" is lost, but at least there is no confusion.

3. 
```
seeds: {
  default: <extimport>,
  others: [<extimport>]
}
```
Here it is very clear what the default is. But I find it a bit cumbersome to separate them between the default and others. It could be a hard decision, and it has to be intentional.

Also, do we allow them to not specify a `default`, but specify `others`? So:
```
seeds: {
  others: [<extimport>]
}
```
which would mean they don't want to have a default, they just have seeds.
Should we allow this if we are going with this? Hm.

4.
```
seeds: {
  default: "devDbSimple",
  fns: [<extimport>]
```
This could be one way to do it. We have `fns` as the main thing, and one can choose one from them to be the default one. `default` would be optional field, `fns` would be required. We would have to check in validation step that `default` matches one of the identifiers in the `fns` field.


I am a bit torn between all of these! You are right that `seeds` being a list makes behaviour of default one being first somewhat unintuitive and therefore change in order can make it confusing. On the other hand, it works nicely when you have just one seed, which is often going to be the case I guess, and also this is something used in development, so it is not like they will be surprised in production with a sudden change. So I am at the end tempted to leave it as it is merely due to how simple it is.

Let's say they don't expect the first element to be run by default. They run `db seed`, they see that first seed was used, and now they know! Maybe that is not terrible?

Reply:
Nice! You will also want to update the changelog comment too I think:

> `wasp db seed` runs the first(default) seed